packages/admin-api-schema/lib/index.js

```diff
@@ -47,39 +55,46 @@ const validate = ({data, version, schema, definitions}) => {
     return jsonSchema.validate(schemaJSON, definitionsJSON, data);
 };
 
-const augmentWithVersion = (version, fn, args) => {
-    return fn(Object.assign({}, {version}, args));
-};
-
-const v2 = {
-    get: (definitionName) => {
-        return augmentWithVersion('v2', get, {definitionName});
-    },
-    list: () => {
-        return augmentWithVersion('v2', list, {});
-    },
-    validate: (options) => {
-        return augmentWithVersion('v2', validate, options);
+const resolveVersion = (version) => {
```

Comment:
It might be nice to have this be a little more specific e.g.
```js
const resolveVersion = (version) => {
  switch(version) {
    case 'canary:
    case 'v3':
      return 'canary';
    case 'v2':
      return 'v2';
    default:
      throw new Error(`Unrecognised version ${version}`);
  }
}
```

I think this is useful because, with the other API, you'd get immediate feedback with a typo, like cannot read property blah of undefined - not a big deal though

Reply:
Maybe a switch is ugly too, but the idea is just to only allow specific strings.

You can jsdoc this too like:
```js
/**
 * @param {'v2'|'v3'|'canary'} version
 */
```

Reply:
The switch/case definitely looks a lot more explicit. Changed implementation to that version. Thanks for the suggestion!